etcdserver/raft.go

```diff
@@ -223,6 +226,8 @@ func (r *raftNode) start(rh *raftReadyHandler) {
 				// gofail: var raftAfterSave struct{}
 
 				if !raft.IsEmptySnap(rd.Snapshot) {
+					// waits etcd server to finish renaming snap db to db.
+					<-replaceDBDone
```

Comment:
The applier will always post to raftDone if there's a snapshot and the raft loop will always wait on raftDone if there's a snapshot. What's to control? I want to avoid the extra channel allocation on this path if possible.